components/mpas-ocean/src/mode_forward/mpas_ocn_time_integration_split.F

```diff
@@ -1917,10 +1919,32 @@ subroutine ocn_time_integrator_split(domain, dt)!{{{
       call mpas_pool_begin_iteration(tracersPool)
       do while ( mpas_pool_get_next_member(tracersPool, groupItr) )
          if (groupItr%memberType == MPAS_POOL_FIELD) then
-            call mpas_dmpar_field_halo_exch(domain, &
-                         groupItr%memberName, timeLevel=2)
+
+            ! Reset iAge to zero where mask == 0
+            if (config_use_idealAgeTracers.and.trim(groupItr%memberName) == 'idealAgeTracers') then
+                call mpas_pool_get_array(tracersPool, &
+                                         groupItr%memberName, &
+                                         tracersGroupNew, 2)
+                call mpas_pool_get_subpool(forcingPool, &
+                                           'tracersIdealAgeFields', &
+                                            tracersIdealAgeFieldsPool)
+                call mpas_pool_get_array(tracersIdealAgeFieldsPool, &
+                                         'idealAgeTracersIdealAgeMask', &
+                                          tracerGroupIdealAgeMask)
+
+                !$omp parallel
+                !$omp do schedule(runtime)
+                do iCell = 1, nCellsAll
+                   tracersGroupNew(1,1,iCell) = tracerGroupIdealAgeMask(1,iCell)*tracersGroupNew(1,1,iCell)
+                end do ! cells
+                !$omp end do
+                !$omp end parallel
+             endif
+
+            call mpas_dmpar_field_halo_exch(domain, groupItr%memberName, timeLevel=2)
```

Comment:
> @maltrud said: looks to me like now only ideal age gets a halo update, instead of all tracers...

This halo update is outside of the ideal age if block. You might have been looking at an old version. But it is hidden, I'll put it at the top with a comment.